### ISSUE TITLE
Allow work to be scheduled after a response has been sent

### ISSUE THREAD

**Issue by BastiDood** [sveltejs/kit#13157]
### Describe the problem

There are cases when we would like to schedule some work after a response has been sent. For my use case, I am writing a Discord bot that performs potentially long-running work. It is thus necessary for me to _acknowledge_ an incoming slash command immediately (i.e., send an HTTP response) and _defer_ a more complete response later (i.e., eventually via the Discord API).

Currently, the Discord API provides an endpoint that lets us [edit the original response](https://discord.com/developers/docs/interactions/receiving-and-responding#edit-original-interaction-response) for a slash command response. Unfortunately, as there is no way to dispatch the work _after_ a response has been sent, there is a fairly common race condition where editing the original response may come first before the original response is even returned from the route handler. Since the slash command response is yet to be received by Discord, invoking the API endpoint for editing the original response (rightly) results in a 404.

### Describe the proposed solution

SvelteKit needs an API for scheduling work _after_ a response has been sent. In the context of my use case, this makes sure that editing the original response comes _after_ the original one has been sent. I believe a solution like the [`after` API in Next.js](https://nextjs.org/docs/app/api-reference/functions/after) is ideal here.

### Alternatives considered

Currently, the way I hacked through this problem is to introduce an artificial timeout. This is far from ideal, of course...

```js
import { json } from '@sveltejs/kit';
import { setTimeout } from 'node:timers/promises';

// +server.js
async function POST(...) {
    // Handle the slash command and preserve some context.
    const { result, ctx } = await handleSlashCommand(...);

    // Detached promise runs in the background.
    void (async () => {
        // HACK: Wait for the response to be sent.
        await setTimeout(1000);

        // Use the existing context to edit the response eventually.
        await doPostResponseWork(ctx, ...);
    })();

    // Send the initial response immediately.
    return json(result);
}
```

### Importance

would make my life easier

**Comment by elliott-with-the-longest-name-on-github** [comment]
I think this would have to be something the adapter would have to implement... waiting until after a request is served isn't possible on all platforms.

**Comment by eltigerchino** [comment]
For reference, we currently have:
- Cloudflare has [waitUntil](https://developers.cloudflare.com/workers/runtime-apis/context/#waituntil) available through `platform.waitUntil`.
https://github.com/sveltejs/kit/blob/96138435fca92632582a529045793acfaf0e70a3/packages/adapter-cloudflare/ambient.d.ts#L7
- Vercel has [waitUntil](https://vercel.com/changelog/waituntil-is-now-available-for-vercel-functions) available through its `@vercel/functions` package.
- Netlify doesn't seem to offer a similar API https://docs.netlify.com/functions/api/
- The Node adapter probably just works if you choose not to await some work while the server still runs after the response is sent.

**Comment by BastiDood** [comment]
> The Node adapter probably just works if you choose not to await some work while the server still runs after the response is sent.

Yup! This is my current workaround, too. Though, I would prefer a cleaner API that hooks into the request-response lifecycle such that the `async` work only gets triggered when the response has finished writing to the socket. The fire-and-forget promise doesn't have that guarantee.
